app/src/main/java/org/mifos/mobile/ui/activities/BiometricAuthentication.kt

```diff
@@ -0,0 +1,106 @@
+package org.mifos.mobile.ui.activities
+
+import android.content.Intent
+import androidx.biometric.BiometricManager
+import android.os.Build
+import android.provider.Settings
+import android.provider.Settings.EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.startActivity
+import androidx.fragment.app.FragmentActivity
+import org.mifos.mobile.R
+
+
+open class BiometricAuthentication(
+    val context: FragmentActivity,
+)  {
+    private val executor = ContextCompat.getMainExecutor(context)
+    private val callback = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        object : BiometricPrompt.AuthenticationCallback() {
+            override fun onAuthenticationFailed() {
+                super.onAuthenticationFailed()
+            }
+
+            override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                super.onAuthenticationSucceeded(result)
+                val intent=Intent(context,HomeActivity::class.java)
+                context.startActivity(intent)
+            }
+
+            override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+                super.onAuthenticationError(errorCode, errString)
+            }
+        }
+    } else {
+        TODO("VERSION.SDK_INT < P")
```

Comment:
Well, what happens if the SDK version is smaller than P? Can we move this logic where we create an object for this class?